cli/src/main/java/org/eclipse/hono/cli/client/ClientConfig.java

```diff
@@ -0,0 +1,95 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.cli.client;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.eclipse.hono.config.ClientConfigProperties;
+import org.springframework.core.MethodParameter;
+import org.springframework.shell.CompletionContext;
+import org.springframework.shell.CompletionProposal;
+import org.springframework.shell.standard.ValueProviderSupport;
+import org.springframework.stereotype.Component;
+
+/**
+ * Configuration class. Holds all the parameters to execute a command.
+ * <p>
+ * A default instance will be created at startup and it will be cloned and modified with custom
+ * parameter for the execution of the command.
+ */
+public class ClientConfig implements Cloneable {
+    public static final String TYPE_TELEMETRY = "telemetry";
+    public static final String TYPE_EVENT = "event";
+    public static final String TYPE_ALL = "all";
+
+    public ClientConfigProperties honoClientConfig;
+
+    public String tenantId;
+    public String deviceId;
+    public String messageType;
+    public int connectionRetryInterval;
+    public int requestTimeoutInSecs;
+    public String messageAddress;
+    public String payload;
```

Comment:
please make these fields private and add accessors, if necessary, in order to increase encapsulation. Otherwise it is very hard to control the state of this class.